src/ckb/types.rs

```diff
@@ -285,6 +297,14 @@ impl Privkey {
     pub fn pubkey(&self) -> Pubkey {
         Pubkey::from(self.0.public_key(secp256k1_instance()))
     }
+
+    pub fn tweak<I: Into<[u8; 32]>>(&self, scalar: I) -> Self {
```

Comment:
Since you only need ref of `[u8; 32]`, `AsRef` is better than `Into` here.

Reply:
The problem I encountered is that https://doc.rust-lang.org/std/convert/trait.AsRef.html#tymethod.as_ref `AsRef<[u8; 32]>` is not automatically implemented for `&[u8; 32]`. The automatic implementation of the rust compiler only works without the length 32. https://doc.rust-lang.org/std/convert/trait.AsRef.html#impl-AsRef%3C%5BT%5D%3E-for-%5BT%5D